File: packages/pip/zwischen/doctor.py
"""Doctor command for Zwischen."""

import shutil
import subprocess

from .installer import get_gitleaks_path, BIN_DIR


def doctor() -> None:
    """Check if required tools are installed."""
    print("\n" + "=" * 60)
    print("Zwischen Doctor - Tool Status")
    print("=" * 60 + "\n")

    all_installed = True

    tools = [
        {
            "name": "gitleaks",
            "description": "Secrets detection",
            "check": get_gitleaks_path,
            "install": "Auto-installed by zwischen init",
        },
        {
            "name": "semgrep",
            "description": "Static analysis (optional)",
            "check": lambda: shutil.which("semgrep"),
            "install": "pip install semgrep",
        },
    ]

    for tool in tools:
        tool_path = tool["check"]()

        if tool_path:
            version = ""
            try:
                result = subprocess.run(
                    [tool_path, "--version"],
                    capture_output=True,
                    text=True,
                )
                version = result.stdout.strip().split("\n")[0]
            except Exception:
                pass

            print(f"\033[32m✓ {tool['name']}\033[0m - {tool['description']}")
            if version:
                print(f"  Version: {version}")
            if tool_path.startswith(str(BIN_DIR)):
                print(f"  Location: {tool_path}")
        else:
            if "optional" not in tool["description"]:
                all_installed = False
            print(f"\033[31m✗ {tool['name']}\033[0m - {tool['description']} - NOT FOUND")
            print(f"  → {tool['install']}")

        print()

    if all_installed:
        print("\033[32m✅ All required tools are installed!\033[0m\n")
    else:
        print('\033[33m⚠️  Some tools are missing. Run "zwischen init" to install them.\033[0m\n')

File: packages/npm/src/doctor.js
const { execSync } = require('child_process');
const { getGitleaksPath, BIN_DIR } = require('./installer');
const path = require('path');

function doctor() {
  console.log('\n' + '='.repeat(60));
  console.log('Zwischen Doctor - Tool Status');
  console.log('='.repeat(60) + '\n');

  let allInstalled = true;

  const tools = [
    {
      name: 'gitleaks',
      description: 'Secrets detection',
      check: () => getGitleaksPath(),
      install: 'Auto-installed by zwischen init'
    },
    {
      name: 'semgrep',
      description: 'Static analysis (optional)',
      check: () => {
        try {
          execSync('which semgrep', { encoding: 'utf8', stdio: ['pipe', 'pipe', 'pipe'] });
          return 'semgrep';
        } catch {
          return null;
        }
      },
      install: 'pip install semgrep'
    }
  ];

  for (const tool of tools) {
    const toolPath = tool.check();

    if (toolPath) {
      let version = '';
      try {
        version = execSync(`${toolPath} --version 2>/dev/null`, { encoding: 'utf8' })
          .trim()
          .split('\n')[0];
      } catch {}

      console.log(`\x1b[32m✓ ${tool.name}\x1b[0m - ${tool.description}`);
      if (version) {
        console.log(`  Version: ${version}`);
      }
      if (toolPath.startsWith(BIN_DIR)) {
        console.log(`  Location: ${toolPath}`);
      }
    } else {
      if (!tool.description.includes('optional')) {
        allInstalled = false;
      }
      console.log(`\x1b[31m✗ ${tool.name}\x1b[0m - ${tool.description} - NOT FOUND`);
      console.log(`  → ${tool.install}`);
    }
    console.log('');
  }

  if (allInstalled) {
    console.log('\x1b[32m✅ All required tools are installed!\x1b[0m\n');
  } else {
    console.log('\x1b[33m⚠️  Some tools are missing. Run "zwischen init" to install them.\x1b[0m\n');
  }
}

module.exports = { doctor };
